backend/api/controllers/PullRequestController.js

```diff
@@ -1,23 +1,63 @@
 const express = require('express')
 const PullRequest = require('../../services/PullRequest')
+const redisClient = require('../../redis')
 
 const PullRequestController = express()
 
 PullRequestController.get('/:year?', async function (req, res) {
   let response = {}
-  let pullRequest = new PullRequest(req.params.year)
+  let year = req.params.year
+  let pullRequest = new PullRequest(year)
 
-  try {
-    response = await pullRequest.getAll()
+  redisClient.exists('users')
+    .then(async (users) => {
+      if (users === 0) {
+        try {
+          response = await pullRequest.getAll()
 
-    response
-      .groupByUser()
-      .sortByMostActive()
+          response
+            .groupByUser()
+            .sortByMostActive()
 
-    res.send(response.data)
-  } catch (error) {
-    res.status(400).send(error)
-  }
+          let arrayOfOrderedUsers = []
+          let data = response.data
+
+          for (let i = 0; i < data.length; i++) {
+            await redisClient.rpush(`pull-requests:${year}:${data[i].username}`,
+              data[i].pullRequests)
+            arrayOfOrderedUsers.push(data[i].pullRequests.length, data[i].username)
+          }
+
+          await redisClient.zadd('users', arrayOfOrderedUsers)
```

Comment:
Some notes here:

1. Since the rank will be per year, it is a good idea to use `users:${year}`
1. You're adding an array instead of items to the set, use `zadd` for each user
1. We don't want to `await` adding each user individually 

`zadd` accepts an incremental score for the the sorting.
In #7 we will have to define a way to create/update this score based on pull-requests count. For now we can just use the sorted array index. Something like this (not tested):

```js
const promises = data.map((item, index) => redisClient.zadd(`users:${year}`, index, item.username))
await Promise.all(promises) //make sure all items are added, but let it be done in parallel

Reply:
I do not understand what you mean in point 2. `zadd` for each user? array instead of items?

Reply:
You are passing `arrayOfOrderedUsers`,
According to redis documentaion, `ZADD` should receive `key`, `score` and `string value` (https://redis.io/commands/zadd).